src/main/java/org/stellar/sdk/Server.java

```diff
@@ -191,8 +193,12 @@ public TransactionsRequestBuilder transactions() {
      * @throws IOException
      */
     public SubmitTransactionResponse submitTransaction(Transaction transaction) throws IOException {
-        if (!this.network.equals(transaction.getNetwork())) {
-            throw new NetworkMismatchException(this.network, transaction.getNetwork());
+        if (!this.network.isPresent()) {
+            this.root();
+        }
```

Comment:
There's a possible race condition when multiple `submitTransaction` are executed at the same time.

Reply:
@bartekn I believe this should be fixed in my latest commit

Reply:
It's better and it works but it's still possible that `root()` will be called twice when `getNetwork()` in the first line of `submitTransaction` returns `null` for 2 or more threads. It's not a big deal because calls to `root()` are cheap so up to you if you want to fix it.

Reply:
yeah, I think that's fine